hydpy/models/whmod/whmod_script.py

```diff
@@ -1720,13 +1790,14 @@ def write_rch_file(
             f"{str(balancefile).rjust(10)}"
             f"         1         1\n"
         )
-        for time_i in data.time:
+        array = data.transpose("time", "row", "col").values
+
+        for timestep in array:
```

Comment:
The variable `array` is unnecessary.  But it could become useful if it had a name or a comment preceding it that would explain what is going on here.  (Except that `data` is of type `DataArray`, I do not see any information explaining it at first sight.)